app/src/test/java/com/hifeful/mealmania/presentation/MealManiaViewModelTest.kt

```diff
@@ -0,0 +1,121 @@
+package com.hifeful.mealmania.presentation
+
+import android.net.Uri
+import android.support.test.runner.AndroidJUnit4
+import com.hifeful.mealmania.presentation.util.DeeplinkResult
+import org.junit.Test
+import org.junit.runner.RunWith
+
+
+@RunWith(AndroidJUnit4::class)
+class MealManiaViewModelTest {
+    private val viewModel = MealManiaViewModel()
+
+    @Test
+    fun `Given mymeals page deeplink, When handleDeeplink is called, Then ShowMyMealsPage DeeplinkResult should be returned`() {
+        // Given
+        val uri = Uri.parse("mealmania://mymeals")
+        
+        // When
+        val actual = viewModel.handleDeeplink(uri)
+
+        // Then
+        assert(actual is DeeplinkResult.ShowMyMealsPage)
+    }
+
+    @Test
+    fun `Given meals search deeplink, When handleDeeplink is called, Then ApplyMealsSearch DeeplinkResult should be returned`() {
+        // Given
+        val uri = Uri.parse("mealmania://search?q=soup")
+
+        // When
+        val actual = viewModel.handleDeeplink(uri)
+
+        // Then
+        assert(actual == DeeplinkResult.ApplyMealsSearch("soup"))
+    }
+
+    @Test
+    fun `Given meals details page deeplink, When handleDeeplink is called, Then ShowMealsPage DeeplinkResult should be returned`() {
+        // Given
+        val uri = Uri.parse("mealmania://meal?id=123")
+
+        // When
+        val actual = viewModel.handleDeeplink(uri)
+
+        // Then
+        assert(actual == DeeplinkResult.ShowMealsPage("123"))
+    }
+
+    @Test
+    fun `Given not supported deeplink, When handleDeeplink is called, Then null should be returned`() {
```

Comment:
`with unsupported deeplink - on handleDeeplink - it should return no deeplink`

Reply:
Changed.